### PR TITLE
Add visual architecture guide (54 Mermaid diagrams) and expand simulation test coverage

### PR DESCRIPTION
**Pull request by ahrav**
## Summary

- Add 13-file Mermaid diagram suite documenting the gossip-rs architecture across
  all 5 boundaries (Identity, Coordination, Shard Algebra, Connector, Persistence)
- Expand simulation tests with convergence proptests, multi-tenant isolation,
  stress tests, and negative regression guards
- Update coordination-testing.md to reflect the expanded test tiers (2 → 6)

### Documentation (13 new files, ~5,000 lines)

New `diagrams/` directory with a self-contained visual architecture guide:

| File | Diagrams | Covers |
|------|----------|--------|
| `00-README.md` | 1 | Index, color legend, reading paths |
| `01-system-overview.md` | 4 | 5-boundary model, crate mapping |
| `02-boundary-dependency-graph.md` | 3 | Type-annotated DAG, anti-patterns |
| `03-id-derivation-dag.md` | 6 | 15-type identity hierarchy |
| `04-end-to-end-scan-flow.md` | 3 | 13-step scan sequence |
| `05-shard-and-run-state-machines.md` | 5 | Shard + Run state machines |
| `06-fencing-protocol.md` | 4 | 5-check zombie prevention |
| `07-lease-lifecycle.md` | 4 | Acquisition, renewal, cursors |
| `08-pagecommit-typestate.md` | 4 | Compile-time state enforcement |
| `09-circuit-breaker.md` | 4 | CB state machine, cascade prevention |
| `10-failure-modes-and-recovery.md` | 6 | 6 failure scenarios |
| `11-tenant-isolation.md` | 5 | 3 isolation layers |
| `12-split-operations.md` | 5 | split_replace + split_residual |
| **Total** | **54** | |

Consistent color coding (B1=blue, B2=green, B3=orange, B4=red, B5=purple)
across all diagrams. Numbered for progressive reading order.

### Simulation Tests (~850 lines changed)

**`harness.rs`**:
- New `SessionTerminalAction` enum with weighted terminal selection
  (complete 50%, park 20%, split_replace 20%, split_residual+complete 10%)
- Centralized `From<*Error> for RejectionKind` conversions
- Extended session lifecycle to exercise `split_replace` and `split_residual` paths
- Improved doc comments throughout

**`mega_sim_tests.rs`** (3 new test categories):
- **Convergence proptests** (`proptest_convergence_sunny`, `proptest_convergence_stormy`) —
  200 cases each asserting bounded liveness (all shards reach terminal state)
- **Multi-tenant isolation** — 2 tenants sharing one coordinator, verifying
  `TenantMismatch` rejection and checker history isolation
- **Stress tests** (`stress_200_shards_stormy`, `stress_split_cascade`) —
  scale and fault-injection edge cases
- **Negative guard** (`zero_liveness_does_not_converge`) — prevents
  `check_convergence` from being hardcoded to `true`

**`coordination-testing.md`**: Updated to document all 6 sub-tiers.

## Test Plan

- [ ] `cargo test` passes (non-`#[ignore]` tests including `multi_tenant_isolation`
  and `zero_liveness_does_not_converge`)
- [ ] `cargo test -- --ignored mega_sim_10k_steps` passes with default seeds
- [ ] Diagram rendering verified on GitHub markdown preview
- [ ] Mermaid blocks render without syntax errors in mermaid.live

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: diagrams/02-boundary-dependency-graph.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+## Cross-References
+
+| Topic | Diagram File |
+|-------|-------------|
+| System overview and five-boundary architecture | `diagrams/01-system-overview.md` |
+| Identity boundary deep-dive | `diagrams/03-identity.md` |
+| Shard algebra operations | `diagrams/04-shard-algebra.md` |
+| Coordination protocol | `diagrams/05-coordination.md` |
+| Connector lifecycle | `diagrams/06-connectors.md` |
+| Persistence guarantees | `diagrams/07-persistence.md` |
```
[the remark this answers is not in the document]
Comment (ahrav) [reply diagrams/02-boundary-dependency-graph.md]: Good catch. All 5 links were pointing to non-existent files. Fixed in 0fc5f5c — also converted the existing line 269 to a clickable relative link for consistency.

File: diagrams/00-README.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+| 04 | `04-end-to-end-scan-flow.md` | 3 | All | 13-step sequence, atomic commit boundary |
+| 05 | `05-shard-and-run-state-machines.md` | 5 | B2 | Shard SM, run SM, splits lifecycle, illegal transitions |
+| 06 | `06-fencing-protocol.md` | 4 | B2 | 5-check validation, zombie resolution, decision tree |
+| 07 | `07-lease-lifecycle.md` | 4 | B2 | Acquisition, renewal timeline, cursor monotonicity |
+| 08 | `08-pagecommit-typestate.md` | 4 | B5 | Typestate SM, partial write failures, compile-time safety |
+| 09 | `09-circuit-breaker.md` | 4 | B4 | CB state machine, cascade prevention, per-connector isolation |
+| 10 | `10-failure-modes-and-recovery.md` | 6 | All | Worker crash, coordinator crash, partitions, split-brain |
+| 11 | `11-tenant-isolation.md` | 5 | B1, B2 | 3 isolation layers, correlation attack, TenantSecretKey |
+| 12 | `12-split-operations.md` | 5 | B2, B3 | split_replace, split_residual, coverage validation |
+| | **Total** | **54** | | |
```
[the remark this answers is not in the document]
Comment (ahrav) [reply diagrams/00-README.md]: Fixed in the same commit (0fc5f5c). Total updated to 55.
